src/common/SourcifyEventManager/SourcifyEventManager.ts

```diff
@@ -4,48 +4,48 @@ import { EventManager, GenericEvents } from "../EventManager";
 interface Events extends GenericEvents {
   "*": (event: string, argument: any) => void;
   "Verification.MatchStored": (match: Match) => void;
-  "Monitor.Error.CantStart": (e: { chainId: string; message: string }) => void;
+  "Monitor.Error.CantStart": (e: { chainId: number; message: string }) => void;
   "Monitor.Started": (obj: {
-    chainId: string;
-    web3url: string;
+    chainId: number;
```

Comment:
I think here it would be better to keep string and do a chainId.toString() when passing it. Just for the sake of always handling the chainId as a string

Reply:
Hmm yeah the only place the `chainId` is a number is at the `Chain` type definition because `chains.json` has it as a number. In the rest of our code it's assumed a string mostly.

https://github.com/ethereum/sourcify/blob/staging/packages/lib-sourcify/src/lib/types.ts#L197

Let's do as you say.